### PR TITLE
Adding a dependabot config, upgrade to PHP 8.0, prepare for 0.4.0

### PR DESCRIPTION
**Pull request by srtfisher**
- Adds a Dependabot configuration.
- Prepares for 0.4.0 release.
- Upgrades to PHP 8.0.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# This file assigns responsibility for reviewing specific files
+# in this repo to specific people or teams.
+# See: https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners
+
+*	@joemcgill
```
Comment (joemcgill) [inline .github/CODEOWNERS]: Fine for now, but wonder if this should be the alley admins group?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Alley Coding Standards
 
-This is a PHPCS ruleset for [Alley Interactive](https://alley.co).
+This is a PHPCS ruleset for [Alley Interactive](https://alley.com).
```
Comment (joemcgill) [inline README.md:3]: ✨

File: README.md [hunk 3]
```diff
@@ -49,42 +49,42 @@ composer run phpcs -- --report=summary
...
 
-```
+```xml
 <?xml version="1.0"?>
 <ruleset>
-	<description>Example project ruleset</description>
-
-    <!-- Include Alley Rules -->
-    <rule ref="Alley-Interactive" />
+  <description>Example project ruleset</description>
```
Comment (joemcgill) [inline README.md:58]: 👾

File: README.md [hunk 4]
```diff
@@ -49,42 +49,42 @@ composer run phpcs -- --report=summary
...
 </ruleset>
 ```
 
-# Change Log
+## Change Log
 
 This project adheres to [Keep a CHANGELOG](https://keepachangelog.com/en/1.0.0/).
 
-## 0.4.0-alpha
+### 0.4.0
```
Comment (joemcgill) [inline README.md:71]: Should probably update this to note that PHPCompat is now configured for PHP 8.0, and note any dependency updates that get applied before we cut a new release.

**Review by joemcgill — APPROVED** [review]
👍🏻 This looks great to me. Thanks for the additions.

File: .github/CODEOWNERS (hunk 1 above)
Comment (srtfisher) [reply .github/CODEOWNERS]: Sames safe enough to me

**Pull request merged**